### ISSUE TITLE
Kan thermostaat niet uitlezen of aansturen via Homebridge

### ISSUE THREAD

**Issue by jorgdeboer** [robertklep/nefit-easy-client#8]
Hi Robert,

Ik heb mijn SD kaart waar homebridge op draaide volledig gewist en weer geflashed met HOOBS. Na de nefit plug-in (v2.2.0) te hebben geïnstalleerd en de config aangepast (zie bijlage), voegt hij inderdaad thermostaat toe.

Echter in zowel de HOOBS omgeving als in Homekit is de thermostaat niet bereikbaar.

Voordat ik mijn SD kaart geflashed had, draaide ik ook al HOOBS met de nefit plug-in en werkte het prima. 

Thanks!

Jorg
[config.txt](https://github.com/robertklep/nefit-easy-client/files/3694747/config.txt)

**Comment by robertklep** [comment]
Zou je eens kunnen proberen met een oudere versie van `homebridge-nefit-easy`?
```
npm i homebridge-nefit-easy@2.1 -g
```

Ik heb vorige week versie 2.2 gepublished en wil even uitsluiten dat daar het probleem zit.

Is je huidige config file een kopie van wat je voorheen gebruikte? Zijn (specifiek) de waarden voor "username", "port" en "pin" veranderd sinds de vorige keer? Mijn ervaring is dat wanneer die waarden hetzelfde blijven, iOS nog wel eens in de war raakt. Als je dan één veld aanpast dan wil dat iOS nog wel eens rechttrekken.

Daarnaast zou je Homebridge kunnen opstarten met -D (--debug), dan zou je meer te zien moeten krijgen over wat de Nefit plugin allemaal onder de motorkap uitvoert.

**Comment by jorgdeboer** [comment]
Ik heb nu 2.1.2 draaien en de “username” van de bridge aangepast. Alles herstart maar nog  geen succes.

**Comment by robertklep** [comment]
Ik ken HOOBS verder niet, maar ik vind het wel vreemd dat die kennelijk ook laat zien dat het niet werkt. Zonder meer info (bv logfiles) kan ik echter ook niet veel doen.

**Comment by jorgdeboer** [comment]
Ik zal morgen even de logfiles sturen.

**Comment by jorgdeboer** [comment]
Zie bijgevoegd de log files, hopelijk kun je er iets mee.
![1](https://user-images.githubusercontent.com/48151659/66291055-1c348380-e8e1-11e9-96f7-2458c8fb07ef.jpg)
![2](https://user-images.githubusercontent.com/48151659/66291056-1c348380-e8e1-11e9-8afb-6c08db4c8800.jpg)

**Comment by robertklep** [comment]
Uit de eerste screenshot blijkt dat het gewoon lijkt te werken, je ziet dat de temperatuur en status wordt opgehaald, en ook dat de temperatuur ingesteld wordt. Dus het probleem zit niet in de Nefit plugin.

Wat ik ook niet helemaal begrijp is dat als het niet werkt, waarom dan wel de temperatuur _ingesteld_ wordt. Want dat kan alleen als je zelf de opdracht daartoe geeft (via iOS of via de HOOBS interface).

**Comment by jorgdeboer** [comment]
eens, echter via de Nefit app zelf werkt het wel om op afstand de temperatuur te bepalen. Ik heb ook al mijn thermostaat gereset en zelfs via Mijn Nefit de gateway verwijderd en weer toegevoegd.

**Comment by robertklep** [comment]
In de eerste log staat dat drie keer de temperatuur is ingesteld via Homekit (naar 11.5, 24 en 25 graden). Ik ga ervan uit dat jij dat deed. Dus de plugin werkt gewoon.

Er is geen verschil tussen "uitlezen" en "instellen" wat dat betreft, als het instellen werkt, dan werkt het uitlezen ook, en dat staat ook in de log: het uitlezen ("Getting current state", "Getting current/target temperature") geeft geen errors.

Waarom iOS dan de temperatuur niet laat zien is een ander probleem, maar dat valt buiten de scope van mijn plugin. Ik heb zelf ook regelmatig vage problemen gehad met Homekit, maar de laatste tijd valt het erg mee (ik gebruik zelf geen Homebridge maar Home Assistant, en daarnaast een oude iPad als Homekit hub).

**Comment by jorgdeboer** [comment]
oke, thanks voor het meedenken in ieder geval!

Ik ga eens naar Home Assistant kijken. lk neem aan dat ik dan via Home Assistant ook de Nefit Easy via Homekit kan benaderen?

**Comment by robertklep** [comment]
Ja, dat kan. Ik moet wel zeggen dat de Nefit Easy implementatie van Home Assistant niet echt geweldig is, als HA de verbinding met de Easy een keer kwijtraakt pikt ie 'em niet automatisch weer op en moet je HA opnieuw opstarten. Maar de Homekit-ondersteuning van HA is wel goed, die werkt voor mij erg stabiel.
